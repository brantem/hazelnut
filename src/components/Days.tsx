import clsx from 'clsx';

import { Day } from 'types/shared';

const Days = ({ days = [] }: { days: Day[] }) => {
  if (!days.length) return null;

  return (
    <div className="space-x-2">
      {days.length === 7 ? (
        <span>Everyday</span>
      ) : (
        days.map((day) => (
          <span key={day} className={clsx({ 'opacity-25': !days.includes(day) })}>
            {day[0]}
          </span>
        ))
      )}
    </div>
  );
};

export default Days;
