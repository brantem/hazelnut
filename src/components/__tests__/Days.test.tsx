import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import Days from 'components/Days';

describe('Days', () => {
  it('should show "Everyday"', () => {
    const { container } = render(
      <Days days={['MONDAY', 'TUESDAY', 'WEDNESDAY', 'THURSDAY', 'FRIDAY', 'SATURDAY', 'SUNDAY']} />,
    );

    expect(container).toMatchSnapshot();
    expect(screen.getByText('Everyday'));
  });

  it('should show only "M"', () => {
    const { container } = render(<Days days={['MONDAY']} />);

    expect(container).toMatchSnapshot();
    expect(screen.getByText('M'));
  });
});
