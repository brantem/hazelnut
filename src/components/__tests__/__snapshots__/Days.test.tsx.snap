// Vitest Snapshot v1

exports[`Days > should render null 1`] = `<div />`;

exports[`Days > should show "Everyday" 1`] = `
<div>
  <div
    class="space-x-2"
    data-testid="days"
  >
    <span>
      Everyday
    </span>
  </div>
</div>
`;

exports[`Days > should show only "M" 1`] = `
<div>
  <div
    class="space-x-2"
    data-testid="days"
  >
    <span
      class=""
    >
      M
    </span>
  </div>
</div>
`;
